lib/usb/usb_standard.c

```diff
@@ -247,19 +248,30 @@ static int usb_standard_set_configuration(usbd_device *usbd_dev,
 					  struct usb_setup_data *req,
 					  uint8_t **buf, uint16_t *len)
 {
-	int i;
+	uint8_t i;
+	const struct usb_config_descriptor *cfg;
 
 	(void)req;
 	(void)buf;
 	(void)len;
 
-	/* Is this correct, or should we reset alternate settings. */
-	if (req->wValue == usbd_dev->current_config) {
-		return 1;
+	if (req->wValue > usbd_dev->desc->bNumConfigurations) {
+		return USBD_REQ_NOTSUPP;
 	}
 
 	usbd_dev->current_config = req->wValue;
 
+	if (usbd_dev->current_config) {
```

Comment:
this is initalizing the new config alternate settings value to 0.
this has to be done every time to a new config (either in user code or ocm3 code).
so, it is better to do in ocm3 because these values are also used by ocm3 too.